### PR TITLE
Use brewed openssl, optional libiconv, and head version.

### PR DESCRIPTION
**Pull request by kkumler**
Bringing over cleaner changes (and closing previous PR #37370).
Since the snapshot is used in master now I left that as-is (instead of devel).

### CODE REVIEW

File: Library/Formula/dcmtk.rb [hunk 1]
```diff
@@ -12,19 +17,22 @@ class Dcmtk < Formula
...
   depends_on "openssl"
+  depends_on "homebrew/dupes/libiconv" => :optional
 
   def install
     ENV.m64 if MacOS.prefer_64_bit?
 
     args = std_cmake_args
     args << "-DDCMTK_WITH_DOXYGEN=YES" if build.with? "docs"
-    args << "-DDCMTK_WITH_OPENSSL=YES" if build.with? "openssl"
+    args << "-DDCMTK_WITH_OPENSSL=YES -DWITH_OPENSSLINC=#{Formula["openssl"].opt_prefix}" if build.with? "openssl"
```
Comment (DomT4) [inline Library/Formula/dcmtk.rb]: It wasn't using our OpenSSL before?

File: Library/Formula/dcmtk.rb [hunk 2]
```diff
@@ -37,6 +45,6 @@ def install
   test do
     system bin/"pdf2dcm", "--verbose",
            test_fixtures("test.pdf"), testpath/"out.dcm"
-    File.exist? testpath/"out.dcm"
+    system bin/"dcmftest", testpath/"out.dcm"
```
Comment (DomT4) [inline Library/Formula/dcmtk.rb]: Why this change?

File: Library/Formula/dcmtk.rb [hunk 3]
```diff
@@ -1,8 +1,13 @@
 class Dcmtk < Formula
   homepage "http://dicom.offis.de/dcmtk.php.en"
+
+  # Current snapshot used for stable now.
   url "http://dicom.offis.de/download/dcmtk/snapshot/dcmtk-3.6.1_20150217.tar.gz"
   sha256 "3cf8f3e52ed8a5240a7facc3a118de411aa54bc9beccba0cf7a975735da35304"
   version "3.6.1-20150217"
+  revision 2
```
Comment (DomT4) [inline Library/Formula/dcmtk.rb]: Why the revision? Only necessary if it wasn't actually using our OpenSSL before. Should be `revision 1` if so.

File: Library/Formula/dcmtk.rb (hunk 1 above)
Comment (kkumler) [inline Library/Formula/dcmtk.rb]: It would build against the system OpenSSL before, even with it depending on the brewed OpenSSL.

File: Library/Formula/dcmtk.rb (hunk 2 above)
Comment (kkumler) [inline Library/Formula/dcmtk.rb]: It gave a higher certainty that the output file was in a valid format instead of just existing.

File: Library/Formula/dcmtk.rb (hunk 1 above)
Comment (DomT4) [inline Library/Formula/dcmtk.rb]: I can't reproduce this, either from bottle or from source.

The source install actively references the correct OpenSSL with the current formula:

```
[ 44%] Building CXX object dcmnet/libsrc/CMakeFiles/dcmnet.dir/dimdump.cc.o
cd /tmp/dcmtk20150521-5874-18qn225/dcmtk-3.6.1_20150217/build/dcmnet/libsrc && /usr/local/Library/ENV/4.3/clang++   -DDCMTK_BUILD_IN_PROGRESS -DUSE_NULL_SAFE_OFSTRING -D_BSD_COMPAT -D_BSD_SOURCE -D_OSF_SOURCE -D_REENTRANT -D_XOPEN_SOURCE_EXTENDED -Ddcmnet_EXPORTS -D_XOPEN_SOURCE_EXTENDED -D_BSD_SOURCE -D_BSD_COMPAT -D_OSF_SOURCE -D_DARWIN_C_SOURCE -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/include/libxml2
```

What's happening for you locally?

File: Library/Formula/dcmtk.rb (hunk 2 above)
Comment (DomT4) [inline Library/Formula/dcmtk.rb]: Fair enough.

File: Library/Formula/dcmtk.rb (hunk 1 above)
Comment (DomT4) [inline Library/Formula/dcmtk.rb]: `otool` lines up as expected as well from-source here.

```
~> otool -L /usr/local/Cellar/dcmtk/3.6.1-20150217/bin/* | grep ssl
	/usr/local/opt/openssl/lib/libssl.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
```

Comment (kkumler) [inline Library/Formula/dcmtk.rb]: My bad. I had that in for the slightly older version/PR but it does look good with this version.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!
